batchflow/models/torch/base.py

```diff
@@ -736,44 +653,116 @@ def make_decay(self, decay, optimizer=None, **kwargs):
+        After it, create training infrastructure (loss, optimizer, decay).
+
+        The order is defined by `order` key in the config, which is [`initial_block`, `body`, `head`] by default.
+        Each item in `order` should describe the block name, the config name and method to create. It can be a:
+            - string, then we use at as name, key and method name
+            - tuple of three elements, which are name, key and method name or callable
+            - dictionary with three items, which are `block_name`, `config_name` and `method`.
+
+        The `block_name` is then used as the identifier in resulting model, i.e. `model.initial_block`, `model.body`.
+        The `config_name` is used to retrieve block creation parameters from config.
+        The `method` is either a string to getattr from the current instance or a callable.
+        Either method or callable should return an instance of nn.Module and accept block parameters.
+        """
+        inputs = inputs or self.make_placeholder_data()
+        inputs = inputs[0] if len(inputs) == 1 else inputs
+
+        blocks = OrderedDict()
+        for item in self.full_config.get('order'):
+            # Get the `block_name`, which is used as the name in the Sequential,
+            #         `config_name`, which is used to retrieve parameters from config,
+            #     and `method`, which is either a callable or name of the method to get from the current instance
+            if isinstance(item, str):
+                block_name = config_name = method = item
+            elif isinstance(item, tuple) and len(item) == 3:
+                block_name, config_name, method = item
+            elif isinstance(item, dict):
+                block_name = item['block_name']
+                config_name = item.get('config_name', block_name)
+                method = item.get('method', config_name)
+
+            # Make block, from the `inputs`, transfer it to device
+            # Important: apply to the `inputs` before showing to the next block, so the shapes/etc are updated
+            block = self.make_block(config_name, method, inputs)
+
+            if block is not None:
+                block.to(self.device)
+                inputs = block(inputs)
+                blocks[block_name] = block
+
+        # Use the OrderedDict in Sequential to give readable names to stages
+        self.model = nn.Sequential(blocks)
+        self.initialize_weights()
+        self.model_to_device()
+
+        self.make_infrastructure()
+
+    def make_block(self, name, method, inputs):
+        """ Create the block with `method` by retrieving its parameters from config by `name`. """
+        config = self.full_config
+        block = config[name]
+
+        if isinstance(block, nn.Module):
+            # Already initialized module
+            pass
+
+        elif isinstance(block, dict):
+            block_params = {**config['common'], **block}
+
+            if 'module' in block_params:
```

Comment:
we also have `base` and `base_block`

Reply:
That is the functionality of `ConvBlock`